native~/Runtime/src/CesiumSimplePlanarEllipsoidCurveImpl.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <CesiumGeospatial/SimplePlanarEllipsoidCurve.h>
+
+#include <DotNet/Unity/Mathematics/double3.h>
+
+#include <memory>
+#include <optional>
+
+namespace DotNet::CesiumForUnity {
+class CesiumSimplePlanarEllipsoidCurve;
+} // namespace DotNet::CesiumForUnity
+
+namespace DotNet::Unity::Mathematics {
+struct double3;
+} // namespace DotNet::Unity::Mathematics
+
+namespace CesiumForUnityNative {
+
+class CesiumSimplePlanarEllipsoidCurveImpl {
+public:
+  CesiumSimplePlanarEllipsoidCurveImpl(
+      const DotNet::CesiumForUnity::CesiumSimplePlanarEllipsoidCurve& path);
+  ~CesiumSimplePlanarEllipsoidCurveImpl();
+
+  bool CreateFromEarthCenteredEarthFixedCoordinates(
+      const DotNet::CesiumForUnity::CesiumSimplePlanarEllipsoidCurve& path,
+      const DotNet::Unity::Mathematics::double3 sourceEcef,
+      const DotNet::Unity::Mathematics::double3 destinationEcef);
+
+  bool CreateFromLongitudeLatitudeHeight(
+      const DotNet::CesiumForUnity::CesiumSimplePlanarEllipsoidCurve& path,
+      const DotNet::Unity::Mathematics::double3 sourceLlh,
+      const DotNet::Unity::Mathematics::double3 destinationLlh);
+
+  DotNet::Unity::Mathematics::double3 GetPosition(
+      const DotNet::CesiumForUnity::CesiumSimplePlanarEllipsoidCurve& path,
+      double percentage,
+      double additionalHeight) const;
+
+private:
+  std::unique_ptr<CesiumGeospatial::SimplePlanarEllipsoidCurve> _curve;
```

Comment:
I noticed one more small thing worth fixing. It would be a bit better, I think, to make this a `std::optional` instead of a `std::unique_ptr`. The former avoids a heap allocation and just has nicer semantics overall (for example, it allows the `CesiumSimplePlanarEllipsoidCurveImpl` to be copyable; not that we're likely to copy it).